File: solutions/ruby/resistor-color-duo/2/resistor_color_duo.rb
# frozen_string_literal: true

module ResistorColorDuo
  RESISTOR_VALUES = %w[
    black brown red orange yellow green blue violet grey white
  ].freeze

  def self.value(input)
    input.first(2).map { |x| RESISTOR_VALUES.index(x) }.join.to_i
  end
end

File: solutions/python/resistor-color-duo/2/resistor_color_duo.py
all_colors = ['black', 'brown', 'red', 'orange', 'yellow', 'green', 'blue', 'violet', 'grey', 'white']

def value(colors):
    first_two = [all_colors.index(c) for c in colors[:2]]
    return first_two[0] * 10 + first_two[1]